projects/fw_102_hw/src/main.c

```diff
@@ -0,0 +1,43 @@
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define COUNTER_TICK_MS 500
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void priv_timer_callback(SoftTimerId timer_id, void *context) {
```

Comment:
private functions should be marked `static`, also by convention we use `prv_` rather than `priv_` :)